File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    my_stack = [s[0]]
    
    for i in range(1, len(s)):
        my_stack.append(s[i])
        if len(my_stack) >= 2 and my_stack[-1] == my_stack[-2]:
            for i in range(2): my_stack.pop()
            
    return 0 if my_stack else 1

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
const solution = (s) => {
    const my_stack = [];
    
    for (const alphabet of s) {
        my_stack.push(alphabet);
        let stack_length = my_stack.length;
        if (stack_length > 1 && my_stack[stack_length-1] === my_stack[stack_length - 2]) {
            my_stack.pop();
            my_stack.pop();
        }
    };
    return my_stack.length === 0 ? 1 : 0; 
}